ablySpec/RealtimeClientConnection.swift

```diff
@@ -1307,6 +1307,35 @@ class RealtimeClientConnection: QuickSpec {
 
             }
 
+            // RTN15
+            context("connection failures once CONNECTED") {
+
+                // RTN15g
+                it("when the connection resume has failed, all channels should be detached with an error reason") {
+                    let options = AblyTests.commonAppSetup()
+                    options.disconnectedRetryTimeout = 1.0
+
+                    let client = ARTRealtime(options: options)
+                    defer { client.close() }
+                    let channel = client.channels.get("test")
+
+                    channel.attach()
+                    expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+
+                    channel.on(.Failed) { errorInfo in
```

Comment:
More generally, every test that uses callbacks should make sure that the callback is being called. Please add that as well.